libr/io/cache.c

```diff
@@ -30,12 +30,16 @@ R_API void r_io_cache_commit(RIO *io, ut64 from, ut64 to) {
 	int ioc = io->cached;
 	io->cached = 2;
 	r_list_foreach (io->cache, iter, c) {
-		if (c->from >= from && c->to <= to) {
-			if (!r_io_write_at (io, c->from, c->data, c->size))
-				eprintf ("Error writing change at 0x%08"PFMT64x"\n", c->from);
-			else 
+		if (from <= c->to - 1 && c->from <= to - 1) {
```

Comment:
so to cover the whole memory space we have to use 0 in from and to, right? weird.. it should be [0, UT64_MAX]

Reply:
and check with >= && <=

Reply:
but that breaks the from+size == to

Reply:
`from == 0 && to == 0` represents [0, 2**64). `from == 0 && to == UT64_MAX` represents [0, 2**64 - 1) .  `from + size == to` stills holds. 

`- 1` still exists, but only when comparing two addresses (because of wraparound). If someone was careless and used `addr < map->to`, it would not break normal code but only wraparound .

Reply:
> and check with >= && <=

This is not an `R_BETWEEN` like comparison. It is used to check whether two half-closed intervals overlap with address space wraparound in mind.

It could be simply `from < c->to && c->from < to` if address space wraparound was not handled.

Reply:
@radare that is exactly why I don't like half-opened, bc they break the mathematical definition of an interval in that scenario, but I also see that ````len = to - from```` is a nice property to have. From a programmer's perspective this is an acceptable tradeoff. But I'm not so sure about the perspective from a user, when you do om and it prints ````0xffffffffffffffff - 0x0```` this could feel weird.

Reply:
```
00400000-00408000 r-xp 00000000 08:02 1844914                            /usr/bin/cat
00607000-00608000 r--p 00007000 08:02 1844914                            /usr/bin/cat
00608000-00609000 rw-p 00008000 08:02 1844914                            /usr/bin/cat
0168d000-016ae000 rw-p 00000000 00:00 0                                  [heap]
```

This is displayed as half-closed. Yes, this is a trade-off. The property `len = to - from` makes programmers a bit harder to make mistakes. Address space wraparound needs extra attention, though.

Reply:
this is broken. And you know it, make it closed